attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -65,6 +65,28 @@ impl Attester for TdxAttester {
         serde_json::to_string(&evidence)
             .map_err(|e| anyhow!("Serialize TDX evidence failed: {:?}", e))
     }
+
+    async fn extend_runtime_measurement(
+        &self,
+        events: Vec<Vec<u8>>,
+        _register_index: Option<u64>,
+    ) -> Result<()> {
+        for event in events {
+            match tdx_attest_rs::tdx_att_extend(&event) {
```

Comment:
The underlying `tdx_att_extend` function is defined in https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/dcap_1.16_reproducible/QuoteGeneration/quote_wrapper/tdx_attest/tdx_attest.c#L477-L479

Do we need to specify a structured input of this function. In other words, will a random byte string work as an input here?

Reply:
@jialez0 I think the commit message should be updated to reflect that this api is implemented only for TDX.

Reply:
@Xynnn007 good point, yes this is the expected struct:
https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/5b320f93e16d0bebf700f2ce4bb5c63c4195bf58/QuoteGeneration/quote_wrapper/tdx_attest/tdx_attest.h#L89-L96

Now the event data itself should also follow a format so that we can verify it.
ACON uses an RTMR event log. @binxing can say more on that. 
I don't think we need that event format in place for this PR, but this is a good reminder that we should start specifying it.

Reply:
Yeah, I think a structured input is definitely necessary, otherwise the caller wouldn't have any idea what `tdx_att_extend` expects.

Maybe a bit off the topic. Have we ever considered keeping measurement logs for RTMRs? With logs, we can support more usages such as measuring the order in which the containers are started, or upgrading containers at runtime.